crates/holochain/src/core/workflow/call_zome_workflow.rs

```diff
@@ -239,9 +239,28 @@ where
         Arc::new(network.clone()),
     ));
 
+    let scratch_records = workspace.source_chain().scratch_records()?;
```

Comment:
Changed this order around to match other validations, so that Deepkey verifies agent validity first and then it's verified on the source chain.

Reply:
Why?

Reply:
To be consistent with how agent validity checks are executed. When DPKI is installed, it is checked first for agent validity, and then the cell's source chain. Before this change it was the other way around. If you attempted to write to the source chain with an invalid agent and had DPKI installed, it would return a source chain error instead of a DPKI error.